src/number_theory/wilson.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2022 John Nicol. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Nicol
+-/
+import number_theory.legendre_symbol.gauss_eisenstein_lemmas
+
+/-!
+# Wilson's theorem.
+
+This file contains a proof of Wilson's theorem.
+
+The heavy lifting is mostly done by the previous `wilsons_lemma`,
+but here we also prove the other logical direction.
+
+This could be generalized to similar results about finite abelian groups.
+
+## References
+
+* [Wilson's Theorem](https://en.wikipedia.org/wiki/Wilson%27s_theorem)
+
+-/
+
+open_locale nat
+
+namespace nat
+
+/-- For `n > 1`, `(n-1)!` is congruent to `-1` modulo `n` only if n is prime. --/
+lemma prime_of_fac_equiv_neg_one
+  {n : ℕ} (h : ((n - 1)! : zmod n) = -1) (h1 : 1 < n) : prime n :=
+begin
+  by_contradiction h2,
+  obtain ⟨m, hm1, hm2 : 1 < m, hm3⟩ := exists_dvd_of_not_prime2 h1 h2,
+  rw [lt_iff_add_one_le, nat.add_le_to_le_sub m h1.le] at hm3,
+  replace hm3 := nat.dvd_factorial (pos_of_gt hm2) hm3,
+  refine hm2.ne' (nat.dvd_one.mp ((nat.dvd_add_right hm3).mp (hm1.trans _))),
+  rw [←zmod.nat_coe_zmod_eq_zero_iff_dvd, cast_add, cast_one, h, add_left_neg],
+end
+
+/-- **Wilson's Theorem**: For `n > 1`, `(n-1)!` is congruent to `-1` modulo `n` iff n is prime. --/
+theorem wilsons_theorem {n : ℕ} (h : 1 < n) :
```

Comment:
Usually mathlib doesn't use the the name of a named theorem, instead opting to put the well-known name in the docstring (like you have) and using a formulaic name. This might be `prime_iff_factorial_eq_neg_one` or something like that.